x/inflation/types/export.go

```diff
@@ -0,0 +1,12 @@
+package types
+
+import (
+	grpc "google.golang.org/grpc"
+)
+
+// GrpcQueryServiceDesc represents the query server's RPC service specification.
+// This gives access to the service name and method names needed for stargate
+// queries.
+func GrpcQueryServiceDesc() grpc.ServiceDesc {
```

Comment:
I'm a little bit worried about exposing the service descriptor as a public variable because it allows for mutations from other places in our codebase, which could mess up the gRPC query server. I believe the query descriptors were made package-private for a reason.

Would it be possible to move the typeURL tests into the packages themselves and not expose the query descriptors? You'll still have access to the list of fully qualified service method names and protobuf names from the `wasmbinding.WasmAcceptedStargateQueries()` function.

Reply:
I'll edit the exported function to return a copy, which will make it more defensive

Reply:
Actually, now that I'm looking at it more closely, the `grpc.ServiceDesc` isn't using pointers, so the underlying value used in the module is immutable, which is what we want.

```go
func GrpcQueryServiceDesc() grpc.ServiceDesc {
	return _Query_serviceDesc
}
```

Since we're exporting as a function, I don't think there's any way to modify the actual `_Query_serviceDesc`